types.d.ts

```diff
@@ -400,6 +400,225 @@ declare class Logger {
     emergency: Log;
 }
 
+/**
+ * MMRP Envelopes are used to encapsulate data to
+ * be sent between MAGE nodes.
+ *
+ * @class MmrpEnvelope
+ */
+declare class MmrpEnvelope {
+    constructor(eventName: string, messages: string | Buffer | string[] | Buffer[], route: string[], returnRoute: string[], flags: any);
+
+    /**
+     *
+     *
+     * @param {string} uri
+     * @param {string} clusterId
+     * @param {any} cb
+     *
+     * @memberof MmrpEnvelope
+     */
+    setMessage(uri: string, clusterId: string, callback: Function): void;
+
+    /**
+     *
+     *
+     * @param {(string | Buffer)} message
+     *
+     * @memberof MmrpEnvelope
+     */
+    addMessage(message: string | Buffer): void;
+
+    /**
+     *
+     *
+     * @param {(string | string[])} route
+     *
+     * @memberof MmrpEnvelope
+     */
+    setRoute(route: string | string[]): void;
+
+    /**
+     *
+     *
+     * @param {string} route
+     *
+     * @memberof MmrpEnvelope
+     */
+    injectRoute(route: string): void;
+
+    /**
+     *
+     *
+     * @param {*} identity
+     *
+     * @memberof MmrpEnvelope
+     */
+    consumeRoute(identity: any): void;
+
+    /**
+     *
+     *
+     * @returns {boolean}
+     *
+     * @memberof MmrpEnvelope
+     */
+    routeRemains(): boolean;
+
+    /**
+     *
+     *
+     * @returns {string}
+     *
+     * @memberof MmrpEnvelope
+     */
+    getFinalDestination(): string;
+
+    /**
+     *
```

Comment:
Returns true if this envelope tracks and contains the sender's route, false otherwise.